hdfs_datanode/tests/test_hdfs_datanode.py

```diff
@@ -29,7 +31,7 @@ def test_check(aggregator, mocked_request):
 
     # Make sure the service is up
     aggregator.assert_service_check(
-        HDFSDataNode.JMX_SERVICE_CHECK, status=HDFSDataNode.OK, tags=HDFS_DATANODE_METRIC_TAGS + CUSTOM_TAGS, count=1
+        HDFSDataNode.JMX_SERVICE_CHECK, status=HDFSDataNode.OK, tags=HDFS_DATANODE_METRIC_TAGS + CUSTOM_TAGS, count=2
```

Comment:
We should change the logic so this only happens once

Reply:
Not sure I follow - think this gets triggered as two service checks because two separate calls are made from the main check?

Reply:
I'm not sure I follow either. This service check was happening only once. Adding the metadata collection logic should not make it happen twice

Reply:
Ah, I see - in this module the service_check gets made as part of the `_rest_request_to_json` call, and that method gets called twice.

In `hdfs_namenode`, that service_check just gets called at the end of the `check` method so its only ever called once.

Updated to mimic the `hdfs_namenode` approach.